cloudfront/wellcomelibrary.org/edge-lambda/src/safeGet.ts

```diff
@@ -0,0 +1,27 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+
+export async function safeGet<T>(
+  url: string,
+  config?: AxiosRequestConfig
+): Promise<Error | T> {
+  const apiResult: AxiosResponse<any> | Error = await axios
+    .get(url, config)
+    .catch((error) => {
+      // Log underlying error here, we don't want to expose it to callers.
+      console.error(error);
```

Comment:
Do you need to log the URL that was associated with this here, or will we be able to match it with the Error block further down?

Reply:
Ah a trick question! Yes - we should log the URL - good spot.